include/qpid/dispatch/router_core.h

```diff
@@ -674,6 +674,15 @@ int qdr_link_process_deliveries(qdr_core_t *core, qdr_link_t *link, int credit);
 
 void qdr_link_flow(qdr_core_t *core, qdr_link_t *link, int credit, bool drain_mode);
 
+/**
+ * The passed in link has been drained, it is no longer in drain mode. Sets the link's drain flag to false and sets credit to core to zero.
+ * Call this right after calling pn_link_drained
+ *
+ * @param core - router core
+ * @param link - the link that has been drained
+ */
+void qdr_link_drained(qdr_core_t *core, qdr_link_t *link);
```

Comment:
I know this is the naming Proton has, but it's confusing.  'qdr_link_drained' looks like a predicate.  Something like 'qdr_link_mark_drained' or '_set_drained' would be a lot clearer IMO.

Reply:
I will change this to qdr_link_set_drained. Thanks.